src/distributed_join.cuh

```diff
@@ -416,25 +578,73 @@ std::unique_ptr<table> distributed_inner_join(
     size_t start_idx = ibatch * mpi_size;
     size_t end_idx   = (ibatch + 1) * mpi_size + 1;
 
+    // temporary buffers for holding compressed data, if needed
+    vector<rmm::device_buffer> compressed_input_left;
+    vector<rmm::device_buffer> compressed_input_right;
+    vector<rmm::device_buffer> compressed_output_left;
+    vector<rmm::device_buffer> compressed_output_right;
+    vector<vector<int64_t>> compressed_recv_offset_left;
+    vector<vector<int64_t>> compressed_recv_offset_right;
+
```

Comment:
But all of them are used outside of `if`?

Reply:
So move `compressed_input_left.clear()` into the if.  And then you can move `compressed_input_left` into the if.

If you do that, then the `clear()` will be called right before the element is destroyed.  But being destroyed will already do a clear, which means that you don't need the clear.

Reply:
That will not work because `all_to_all_comm_with_compression` might be asynchronous. It's only safe to reclaim the memory after  `if (communicator->group_by_batch()) communicator->stop();`

Reply:
Ah, okay.  That makes sense.  I guess you can leave it like it is.  Two alternatives:

1. Put a comment to explain this so that the next person looking at the code will understand why you are calling clear and why it has to be after `stop`, or
2. Just don't call `clear`.  Are you actually running out of memory?  It's going to get cleaned up at the end of the function anyway and the fact that you clear some but not all of them looks weird.

Whatever you choose seems fine.

Reply:
> Put a comment to explain this so that the next person looking at the code will understand why you are calling clear and why it has to be after stop

Yeah I added a comment to explain that.

> Are you actually running out of memory?

Yes. Being able to run as large a problem as possible is critical for us.